ts-client/src/amm/tests/pool.test.ts

```diff
@@ -100,7 +96,52 @@ describe('Pool', () => {
       tokenBAmount,
       tradeFeeBps,
     );
+
+    pool = await AmmImpl.create(provider.connection, _pool, wsolTokenInfo, usdcTokenInfo);
   });
 
-  it('should able to subscribe reserve changes', async () => {});
+  it('should able to subscribe reserve changes', async () => {
+    const poolState = (await ammProgram.account.pool.fetchNullable(pool.address)) as any as PoolState;
+    const aVaultLp = poolState.aVaultLp;
+    const bVaultLp = poolState.bVaultLp;
+
+    const aVaultLpChangeSubId = provider.connection.onAccountChange(aVaultLp, (accountInfo, context) => {
+      console.log('Vault A LP changed');
```

Comment:
can you do pool.refresh() and getSwapQuote in this function?

Reply:
updated

Reply:
no, I mean you also push getSwapQuote in this function (no need to add swapTx) and console.log the swapOutAmount